controller.go

```diff
@@ -266,7 +270,27 @@ func updateAlarmDefinition(id string, r Resource) error {
 	definitionRequest := convertToADRequest(r.Spec.AlarmDefinitionElement)
 	result, err := monascaclient.PatchAlarmDefinition(id, definitionRequest)
 	if err != nil {
-		return err
+		if !strings.HasPrefix(err.Error(), "Error: 422") {
+			return err
+		}
+		// Some updates are not allowed, so try deleting and recreating instead
+		// NOTE: this will remove the history from the alarm definition
```

Comment:
More correctly: This will remove all alarms using this alarm definition which will remove any history